.misc/build_and_test.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+
+# Get environment variables, readily decrypted by rultor
+#source ../rultor_secrets.sh
+
+if [${UNSUPPORTED} == "false"] || [${TRAVIS_BRANCH} == "master"]; then
```

Comment:
Below the if-conditions effectively form an AND/`&&`, now you use OR/`||`. Is this the same?

Reply:
Yup, it is the same

Reply:
can you explain me why? :) Because I still believe this isn't the same :)

Reply:
Here's why I think it's equivalent:

1) In the previous commit we had this:

	   if [ ${UNSUPPORTED} == "true"]; then
	     if [ ${TRAVIS_BRANCH} == "master" ]; then
	       py.test;
	       python setup.py bdist_wheel;
	       pip install ./dist/coala-*.whl;
	       pip install coala-bears[alldeps] --pre -U;
	     fi
	   else
	     py.test;
	     python setup.py bdist_wheel;
	     pip install ./dist/coala-*.whl;
	     pip install coala-bears[alldeps] --pre -U;
	   fi

there are two conditions, 2^n cases could happen

	unsupported == true | 0 0 1 1
	branch == master    | 0 1 0 1

As you can see 
in (0, 0) (python >= 3.4, not master) we go staight to `else`, i. e. setup & run tests,
in (0, 1) (python >= 3.4, master) we again go to else, setup & run tests,
in (1, 0) (python < 3.4, not master) don't do anything and 
in (1, 1) (python < 3.4, master) we go to first build-and-test block

I think it needs refactoring, but still corresponds with somethings you would want to see.

2) I the last commint the previous code was changed to:

		if [${UNSUPPORTED} == "false"] || [${TRAVIS_BRANCH} == "master"]; then
		    cd ..
		    py.test;
		    python setup.py bdist_wheel;
		    pip install ./dist/coala-*.whl;
		    pip install coala-bears[alldeps] --pre -U;

		    # https://github.com/coala/coala-bears/issues/1037
		    if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
		      sed -i.bak '/bears = GitCommitBear/d' .coafile
		    fi

		    coala --non-interactive
		    python setup.py docs
		fi

let's create the same table for this predicate (btw, can I call it a predicate? xD)

	unsupported == true | 0 0 1 1
	branch == master    | 0 1 0 1

This is slightly annoying, because we changed unsupported == true to the opposite, 
but still:
(0, 0) (not unsupported, not master) -> build & test
(0, 1) (not unsupported, master)     -> build & test
(1, 0) (unsupported, not master) 	 -> do nothing
(1, 1) (unsupported, master) 		 -> build & test


So, in my point of view those two pieces code are equivalent, the only thing, 
that differs is that I moved running coala and generating docs inside the if,
because I thought, that if we don't need to run tests, we don't need docs.
I can reverse it, if you want it back.

Reply:
Also, look at my reply to RaiVaibhav :)

Can we drop building and buidling docs on python < 3.4?

Reply:
Indeed, good old boolean functions.

> Can we drop building and buidling docs on python < 3.4?

File an issue ;)